src/System.IO.Compression.ZipFile/src/System/IO/Compression/ZipFile.cs

```diff
@@ -504,38 +505,45 @@ public static void ExtractToDirectory(String sourceArchiveFileName, String desti
                 // to be greater than the length of typical entry names from the file system, even
                 // on non-Windows platforms. The capacity will be increased, if needed.
                 const int DefaultCapacity = 260;
-                char[] entryNameBuffer = new char[DefaultCapacity];
+                char[] entryNameBuffer = ArrayPool<char>.Shared.Rent(DefaultCapacity);
```

Comment:
Ah good catch, I missed EnsureCapacity. I'll change that function to return the buffer and rent a new, bigger one if necessary